### PR TITLE
Add New Product Facades

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, we use template refs to access containers which are essentially bad versions of facades.

Fixes: N/A

Touches: #294 

## What is the new behavior?
This adds the initial facade concept. Implementation will come in a later patch.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/product/src/facades/best-sellers/best-sellers.facade.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  });
+
+  it('should be created', () => {
+    expect(facade).toBeTruthy();
+  });
+
+  it('should be able to dispatch an action to the store', () => {
+    spyOn(store, 'dispatch');
+    facade.dispatch({ type: 'SOME_TYPE' });
+    expect(store.dispatch).toHaveBeenCalledTimes(1);
```
Comment (lderrickable) [inline libs/product/src/facades/best-sellers/best-sellers.facade.spec.ts]: I think you should do:
```
let action = {type: 'SOME_TYPE'};
facade.dispatch(action);
expect(store.dispatch).toHaveBeenCalledWith(action);
```
just to have a slightly more useful test

File: libs/product/src/facades/product-grid/product-grid.facade.spec.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  });
+
+  it('should be created', () => {
+    expect(facade).toBeTruthy();
+  });
+
+  it('should be able to dispatch an action to the store', () => {
+    spyOn(store, 'dispatch');
+    facade.dispatch({ type: 'SOME_TYPE' });
+    expect(store.dispatch).toHaveBeenCalledTimes(1);
```
Comment (lderrickable) [inline libs/product/src/facades/product-grid/product-grid.facade.spec.ts]: And here (see other comment)

File: libs/product/src/facades/product/product.facade.spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  });
+
+  it('should be created', () => {
+    expect(facade).toBeTruthy();
+  });
+
+  it('should be able to dispatch an action to the store', () => {
+    spyOn(store, 'dispatch');
+    facade.dispatch({ type: 'SOME_TYPE' });
+    expect(store.dispatch).toHaveBeenCalledTimes(1);
```
Comment (lderrickable) [inline libs/product/src/facades/product/product.facade.spec.ts]: And here (see other comment)

File: libs/product/src/reducers/index.ts [hunk 4]
```diff
@@ -131,3 +131,9 @@ export const selectBestSellersIdsState : MemoizedSelector<object, string[]> = cr
   selectBestSellersState,
   fromBestSellers.getBestSellersIds
 );
+
+export const selectBestSellersProducts: MemoizedSelector<object, DaffProduct[]> = createSelector(
```
Comment (lderrickable) [inline libs/product/src/reducers/index.ts]: needs a test

File: libs/product/src/facades/best-sellers/best-sellers.facade.spec.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/product/src/facades/best-sellers/best-sellers.facade.spec.ts]: addressed in 3185d1be860c060dac3f846dca340491ae2aff1a

File: libs/product/src/facades/product-grid/product-grid.facade.spec.ts (hunk 2 above)
Comment (damienwebdev) [reply libs/product/src/facades/product-grid/product-grid.facade.spec.ts]: addressed in 3185d1be860c060dac3f846dca340491ae2aff1a

File: libs/product/src/facades/product/product.facade.spec.ts (hunk 3 above)
Comment (damienwebdev) [reply libs/product/src/facades/product/product.facade.spec.ts]: addressed in 3185d1be860c060dac3f846dca340491ae2aff1a

File: libs/product/src/reducers/index.ts (hunk 4 above)
Comment (damienwebdev) [reply libs/product/src/reducers/index.ts]: addressed in d664048f86694c3951420b278205f1f6ca68860e

**Pull request merged**